src/components/guestsList/guestsList.js

```diff
@@ -0,0 +1,243 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { TableVariant } from '@patternfly/react-table';
+import { helpers } from '../../common';
+import { apiQueries, connect, reduxActions, reduxSelectors } from '../../redux';
+import { Loader } from '../loader/loader';
+import { inventoryListHelpers } from '../inventoryList/inventoryListHelpers';
+import { RHSM_API_QUERY_TYPES } from '../../types/rhsmApiTypes';
+import { Table } from '../table/table';
+
+/**
+ * A system inventory guests component.
+ *
+ * @augments React.Component
+ * @fires onUpdateGuestsData
+ * @fires onScroll
+ */
+class GuestsList extends React.Component {
+  state = { currentPage: 0, limit: 10, previousData: [] };
+
+  componentDidMount() {
+    this.onUpdateGuestsData();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { currentPage } = this.state;
+
+    if (currentPage !== prevState.currentPage) {
+      this.onUpdateGuestsData();
+    }
+  }
+
+  /**
+   * Call the RHSM APIs, apply filters.
+   *
+   * @event onUpdateGuestsData
+   */
+  onUpdateGuestsData = async () => {
+    const { currentPage, limit } = this.state;
+    const { getHostsInventoryGuests, query, id } = this.props;
+
+    if (id) {
+      const updatedQuery = {
+        ...query,
+        [RHSM_API_QUERY_TYPES.LIMIT]: limit,
+        [RHSM_API_QUERY_TYPES.OFFSET]: currentPage * limit || 0
+      };
+
+      const { guestsQuery } = apiQueries.parseRhsmQuery(updatedQuery);
+      await getHostsInventoryGuests(id, guestsQuery);
+    }
+  };
+
+  /**
+   * Update page state.
+   *
+   * @event onScroll
+   * @param {object} event
+   */
+  onScroll = event => {
+    const { target } = event;
+    const { currentPage, limit, previousData } = this.state;
+    const { numberOfEntries, pending, listData } = this.props;
+
+    const bottom = target.scrollHeight - target.scrollTop === target.clientHeight;
+
+    if (numberOfEntries > (currentPage + 1) * limit && bottom && !pending) {
+      const newPage = currentPage + 1;
+      const updatedData = [...previousData, ...(listData || [])];
+
+      this.setState({
+        previousData: updatedData,
+        currentPage: newPage
+      });
+    }
+  };
+
+  renderLoader() {
```

Comment:
Optional:
As a possible 2nd iteration task it may be useful to create a "scrollable table" component